src/lualib/ArrayReduce.ts

```diff
@@ -1,20 +1,23 @@
+/** @vararg */
+interface VarArg<T> extends Array<T> {}
+
 // https://www.ecma-international.org/ecma-262/9.0/index.html#sec-array.prototype.reduce
 function __TS__ArrayReduce<T>(
     this: void,
     arr: T[],
     callbackFn: (accumulator: T, currentValue: T, index: number, array: T[]) => T,
-    initial?: T
+    ...initial: VarArg<T>
 ): T {
     const len = arr.length;
 
-    if (len === 0 && initial === undefined) {
+    if (len === 0 && select("#", ...initial) === 0) {
         // tslint:disable-next-line: no-string-throw
         throw "Reduce of empty array with no initial value";
     }
 
     let k = 0;
-    let accumulator = initial;
-    if (initial === undefined) {
+    let accumulator = select(1, ...initial);
+    if (accumulator === undefined) {
```

Comment:
It doesn't look like that would fix that issue. The test case doesn't cover it, but:

```ts
['a', 'b'].reduce((a, b) => console.log(a, b));
// "a" "b"
['a', 'b'].reduce((a, b) => console.log(a, b), undefined);
// undefined "a"
// undefined "b"
```